AthleteIO/src/ro/siit/Athlete/AthleteRepository.java

```diff
@@ -0,0 +1,129 @@
+package ro.siit.Athlete;
+
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class AthleteRepository {
+
+    private static List<Athlete> athleteList = new ArrayList<>();
+
+    /**
+     * This method reads a file, transfers the data to create objects of type Athlete and adds them to a collection of type ArrayList.
+     *
+     * @param fileName
+     * @return a collection of type ArrayList of objects of type Athlete.
+     * @throws IOException if file not found
+     */
+
+    public static List readFromFileToList(String fileName) throws IOException {
+
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(new FileInputStream(fileName)))) {
+            String s;
+
+            while ((s = reader.readLine()) != null) {
+                if ((s.length() > 0)) {
+                    String[] athletesArr = s.split(",");
+                    athleteList.add(new Athlete(Integer.parseInt(athletesArr[0]), athletesArr[1], athletesArr[2], athletesArr[3], athletesArr[4], athletesArr[5], athletesArr[6]));
+                }
+            }
+            System.out.println("Following athletes were read from the file:");
+            for (Athlete a : athleteList) {
+                System.out.println(a);
+            }
+        }
+        return athleteList;
+    }
+
+    /**
+     * This method takes the Athlete objects stored in an ArrayList and calculates the actual time.
+     * This also takes into account the penalty time calculated from the missed shots.
+     *
+     * @param athletesList
+     */
+    public static void calculateTime(List<Athlete> athletesList) {
```

Comment:
Metoda care calculeaza timpul tinand cont de penalizari putea fi incapsulata in clasa `Athlete`.